analysis/analysis.py

```diff
@@ -34,6 +34,96 @@ def load_data_csv(csv_path: Path):
     return out_data
 
 
+def fetch_responses(dataset):
```

Comment:
can you give a more descriptive function name (this does a more specific thing than what this implies, and has to live in a codebase with a lot of other function names...)

Reply:
Fixed - renamed function `extract_responses_by_context`